src/core/BaseNode.js

```diff
@@ -245,5 +178,72 @@ export default Class(null, {
    * @method exit
    * @param {Tick} tick A tick instance.
    **/
-  exit: function(tick) {},
-});
+  exit(tick) {}
+};
+
+/**
+ * Node ID.
+ * @property {string} id
+ * @readonly
+ **/
+BaseNode.prototype.id = null;
```

Comment:
Hmm, but we're never changing or accessing that `BaseNode.prototype.id` value directly, or did I miss something?
When creating new instance of BaseNode (or anything that extends from it), its constructor creates new property `id` for that instance. So after new instance has been created, calling `newNode.id` will give you that instance's id, instead of the one from the prototype.